core/ir/Input.cpp

```
@@ -40,6 +40,13 @@ bool valid_dtype_format_combo(nvinfer1::DataType dtype, nvinfer1::TensorFormat f
         default:
           return false;
       }
+    case nvinfer1::DataType::kBOOL: // Supports Linear (NCHW)
+      switch (format) {
```

We probably should be more clear to the user why a BOOL / NHWC combo doesnt work

---

NHWC is only bound to FP32 according to https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/namespacenvinfer1.html#ac3e115b1a2b1e578e8221ef99d27cd45

---

Yeah, I understand, I mean we need to throw a descriptive error or something

---

Probably put this at the API level